pennylane/operation.py

```diff
@@ -2048,20 +2048,18 @@ def _unflatten(cls, data, _):
         return cls(*data)
 
     def __init__(self, *args):  # pylint: disable=super-init-not-called
-        wires = [op.wires for op in args]
-        if len(wires) != len(set(wires)):
-            warnings.warn(
-                "Tensor object acts on overlapping wires; in some PennyLane functions this will lead to undefined behaviour",
-                UserWarning,
-            )
-
```

Comment:
I'd prefer to leave this during `__init__`, since it doesn't really have anything to do with queuing.  If the problem is `Tensor.queue` mutating the operator, we can just make it so that `Tensor.queue` no longer mutates the operator?